### PR TITLE
Update mypy to latest version

### PR DESCRIPTION
**Pull request by ezdac**
This updates the mypy dependency to
the latest version and will include typing changes
to comply with the latest version.

This will also fix the phony pre-commit error (see #6689)

### CODE REVIEW

File: raiden/tests/utils/smoketest.py [hunk 1]
```diff
@@ -511,7 +512,7 @@ def setup_smoketest(
     print_step: StepPrinter,
     free_port_generator: Iterator[Port],
     debug: bool = False,
-    stdout: IO = None,
```
Comment (karlb) [inline raiden/tests/utils/smoketest.py:514]: Why is this necessary?

File: raiden/transfer/mediated_transfer/mediation_fee.py [hunk 2]
```diff
@@ -79,11 +79,13 @@ def _collect_x_values(
...
     To find the penalty fee for the current transfer.
     """
-    all_x_vals = [x - balance_in for x in penalty_func_in.x_list] + [
+    all_x_vals: List[Fraction] = [x - balance_in for x in penalty_func_in.x_list] + [
         balance_out - x for x in penalty_func_out.x_list
     ]
-    limited_x_vals = (max(min(x, balance_out, max_x), 0) for x in all_x_vals)
-    return sorted(set(Fraction(x) for x in limited_x_vals))
+    limited_x_vals = (
+        max(min(x, Fraction(balance_out), Fraction(max_x)), Fraction(0)) for x in all_x_vals
```
Comment (karlb) [inline raiden/transfer/mediated_transfer/mediation_fee.py:86]: `min` and `max` between `int`s and `Fraction`s should be a valid, safe operation. `Fraction` is not just a mypy type cast, but a real python constructor, so I'm afraid this might slow down these operations. This is called repeatedly during the mediation fee calculation, so performance can be relevant here. If mypy mistakenly complains about a `max(int, Fraction)`, consider using a type ignore.

File: raiden/ui/cli.py [hunk 3]
```diff
@@ -505,7 +505,7 @@ def options(func: Callable) -> Callable:
         ),
     ]
 
-    if importlib.util.find_spec("IPython"):
+    if find_spec("IPython"):
```
Comment (karlb) [inline raiden/ui/cli.py:508]: I prefer the qualified name for such rare functions, but it's fine either way.

File: raiden/utils/mediation_fees.py [hunk 4]
```diff
@@ -16,7 +16,9 @@ def ppm_fee_per_channel(per_hop_fee: ProportionalFeeAmount) -> ProportionalFeeAm
     for how to get to this formula.
     """
     per_hop_ratio = Fraction(per_hop_fee, 10 ** 6)
-    return ProportionalFeeAmount(round(per_hop_ratio / (per_hop_ratio + 2) * 10 ** 6))
+    return ProportionalFeeAmount(
+        round(Fraction(per_hop_ratio / (per_hop_ratio + 2) * 10 ** 6))  # type: ignore
```
Comment (karlb) [inline raiden/utils/mediation_fees.py:20]: `round(Fraction(value))` brings no improvement over `round(value)`. Why the change here?

If you want to use Fraction to increase the accuracy, you need to use `Fraction(num, den)`. But when rounding anyway, I don't see the benefit for that.

File: raiden/tests/utils/smoketest.py (hunk 1 above)
Comment (ezdac) [reply raiden/tests/utils/smoketest.py:514]: Sorry, I now remember the section in the raiden dev-guide about default-`None` typing.

File: raiden/ui/cli.py (hunk 3 above)
Comment (ezdac) [reply raiden/ui/cli.py:508]: Somehow mypy does not want to accept the qualified name, even with a `# type: ignore`.
Maybe some "strict" mypy setting?
I would prefer it this way, so that mypy is happy and our pre-commit hooks don't fail constantly because of this.

File: raiden/transfer/mediated_transfer/mediation_fee.py (hunk 2 above)
Comment (ezdac) [reply raiden/transfer/mediated_transfer/mediation_fee.py:86]: Mypy was happy, I didn't think about performance implications.
But you're right, it's a type-stub issue, and type conversion can be bad here.

File: raiden/utils/mediation_fees.py (hunk 4 above)
Comment (ezdac) [reply raiden/utils/mediation_fees.py:20]: Sorry, this was a leftover being there despite the type-ignore

File: raiden/ui/cli.py (hunk 3 above)
Comment (karlb) [reply raiden/ui/cli.py:508]: That's a strange kind of error. But this is totally unimportant, so let's ignore it.

**Review by karlb — APPROVED** [review]
